src/extension/settings/menu-swaps.tsx

```diff
@@ -0,0 +1,285 @@
+    refundSwapFlowRepository,
+  } = useStorageContext();
+
+  const [error, setError] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [refundableSwapParams, setRefundableSwapParams] = useState<RefundableSwapParams>();
+  const [touched, setTouched] = useState(false);
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const network = cache?.network ?? 'liquid';
+  const boltz = new Boltz(boltzUrl[network], networks[network].assetHash, zkpLib);
+
+  useEffect(() => {
+    (async () => {
+      const params = await refundSwapFlowRepository.getParams();
+      if (params) {
+        setTouched(true);
+        setRefundableSwapParams(params);
+        await refundSwapFlowRepository.reset();
+        document.getElementById('json')!.innerText = JSON.stringify(params);
+        void validateParams(JSON.stringify(params));
+      }
+    })().catch(console.error);
+  }, []);
+
+  const findDerivationPath = async (refundPublicKey: string): Promise<string> => {
+    // get account
+    const accountFactory = await AccountFactory.create(walletRepository);
+    const accountName = network === 'liquid' ? MainAccount : MainAccountTest;
+    const mainAccount = await accountFactory.make(network, accountName);
+    const addresses = await mainAccount.getAllAddresses();
+    // find address
+    const [usedAddress] = addresses.filter((a) => a.publicKey === refundPublicKey);
+    if (usedAddress?.derivationPath) return usedAddress.derivationPath;
+    // check if by any chance the address used is missing on marina cache
+    // by iterating through the next 20 addresses
+    for (let i = 0; i < 20; i++) {
+      const nextAddress = await mainAccount.getNextAddress(false);
+      if (nextAddress.publicKey === refundPublicKey && nextAddress.derivationPath)
+        return nextAddress.derivationPath;
+    }
+    throw new Error('derivation path not found');
+  };
+
+  const getKeyPairFromDerivationPath = async (
+    derivationPath: string,
+    baseDerivationPath: string,
+    password: string
+  ): Promise<ECPairInterface> => {
+    const encrypted = await walletRepository.getEncryptedMnemonic();
+    if (!encrypted) throw new Error('No mnemonic found in wallet');
+    const decryptedMnemonic = await decrypt(encrypted, password);
+    const masterNode = bip32.fromSeed(await mnemonicToSeed(decryptedMnemonic));
+    const key = masterNode
+      .derivePath(baseDerivationPath)
+      .derivePath(derivationPath.replace('m/', ''));
+    return ECPairFactory(ecc).fromPrivateKey(key.privateKey!);
+  };
+
+  const getBlockTip = async (): Promise<number> => {
+    const webExplorerURL = await appRepository.getWebExplorerURL();
+    const url = `${webExplorerURL?.replace(
+      // stupid bug from mempool
+      'liquid.network/testnet',
+      'liquid.network/liquidtestnet'
+    )}/api/blocks/tip/height`;
+    const { data } = await axios.get(url);
+    return data;
+  };
```

Comment:
does not Electrum/ChainSource give this info?

Reply:
No :(